Week1/homework/world-queries.txt

```diff
@@ -1,40 +1,250 @@
+| Australia          |
+| India              |
+| Argentina          |
+| Kazakstan          |
++--------------------+
+10 rows in set (0.00 sec)
+
+OR..
+mysql> SELECT name FROM country GROUP BY SurfaceArea DESC LIMIT 10;
++--------------------+
+| name               |
++--------------------+
+| Russian Federation |
+| Antarctica         |
+| Canada             |
+| China              |
+| United States      |
+| Brazil             |
+| Australia          |
+| India              |
+| Argentina          |
+| Kazakstan          |
++--------------------+
+10 rows in set (0.04 sec)
+--------------------------------------------------------------
+
+5.What's the top 10 most populated cities?
+
+mysql> SELECT name FROM city ORDER BY population DESC LIMIT 10;
++-----------------------+
+| name                  |
++-----------------------+
+| Mumbai (Bombay)       |
+| Seoul                 |
+| S�?�o Paulo           |
+| Shanghai              |
+| Jakarta               |
+| Karachi               |
+| Istanbul              |
+| Ciudad de M�?�xico    |
+| Moscow                |
+| New York              |
++-----------------------+
+10 rows in set (0.01 sec)
+--------------------------------------------------------------
+
+6.Select the top 3 countries by population that their names start with the letter �P�
+
+mysql> SELECT name FROM country WHERE name LIKE 'P%' ORDER BY population DESC LIMIT 3;
++-------------+
+| name        |
++-------------+
+| Pakistan    |
+| Philippines |
+| Poland      |
++-------------+
+3 rows in set (0.00 sec)
+--------------------------------------------------------------
+
+7.What is the population in Asia?
+
+mysql> SELECT sum(population) as 'Popluation of Asia' FROM country WHERE continent='Asia';
++--------------------+
+| Popluation of Asia |
++--------------------+
+|         3705025700 |
++--------------------+
+1 row in set (0.07 sec)
+--------------------------------------------------------------
+
+8.Which languages are spoken in the Region of �South America�
+
+mysql> SELECT distinct language FROM countrylanguage as cl JOIN country as c
+    -> on cl.countrycode=c.code
+    -> WHERE c.region='South America';
++------------------+
+| language         |
++------------------+
+| Indian Languages |
+| Italian          |
+| Spanish          |
+| Aimar�?�         |
+| Guaran�?�        |
+| Ket�?�ua         |
+| German           |
+| Japanese         |
+| Portuguese       |
+| Araucan          |
+| Rapa nui         |
+| Arawakan         |
+| Caribbean        |
+| Chibcha          |
+| Creole English   |
+| English          |
+| Creole French    |
+| Hindi            |
+| Sranantonga      |
+| Goajiro          |
+| Warrau           |
++------------------+
+21 rows in set (0.05 sec)
+--------------------------------------------------------------
+
+9.What are the languages spoken on all cities named �Barcelona� (you may need to join 3 tables =) )
+
+mysql> SELECT distinct language FROM countrylanguage as cl JOIN city as c

```

Comment:
This works as advertised 👏 ... but part of the fun was realising there was more than one city named Barcelona ! 
I was curious so i gave this one a try !
```sql
SELECT
  ci.name,
  co.name,
  l.language
FROM country co
  JOIN city ci on ci.countrycode = co.Code
  JOIN countrylanguage l ON l.CountryCode = co.Code
WHERE ci.Name = 'Barcelona';
```